sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -369,29 +371,46 @@ private void createRunnerAndConsumersForPTransformRecursively(
           .setProcessBundleProgress(BeamFnApi.ProcessBundleProgressResponse.getDefaultInstance());
     }
 
-    // TODO(BEAM-6597): This should really only be reporting monitoring infos where the data
-    // changed
-    // and we should be using the short id system.
+    Map<String, ByteString> monitoringData = monitoringData(bundleProcessor);
+    if (runnerAcceptsShortIds) {
+      response.putAllMonitoringData(monitoringData);
+    } else {
+      for (Map.Entry<String, ByteString> metric : response.getMonitoringDataMap().entrySet()) {
```

Comment:
for (Map.Entry<String, ByteString> metric : monitoringData.entrySet()) ?

Reply:
Done.